### ISSUE TITLE
rocm-smi reporting a non-existant GPU 0?

### ISSUE THREAD

**Issue by seibert** [RadeonOpenCompute/ROCm#561]
On my system with ROCm 1.9, I see the following output from `rocm-smi`:
```
====================    ROCm System Management Interface    ====================
================================================================================
 GPU  Temp    AvgPwr   SCLK     MCLK     Fan      Perf    SCLK OD    MCLK OD
  0   N/A     N/A      N/A      N/A      0%       N/A       N/A        N/A
  1   55c     15.0W    852Mhz   167Mhz   10.98%   auto      0%         0%
================================================================================
====================           End of ROCm SMI Log          ====================
```

Is the non-existent GPU 0 an artifact of the CPU being enumerated as an HSA agent?  Might be less confusing to filter it out of rocm-smi output.

**Comment by jlgreathouse** [comment]
`rocm-smi` walks through all of the GPUs in the system shown in, for example, `/sys/class/drm/#/`. The DRM system can include non-AMD GPUs. For example, d you have an integrated GPU in your laptop or server motherboard?

**Comment by kentrussell** [comment]
Thanks @jlgreathouse , that's exactly what we do. I have been working forever on trying to only report GPUs that actually matter and have topology loaded, but it hasn't been done yet. For now, we just go through /sys/class/drm/cardX , and X is the GPU that we use. So in your case, it would be an integrated graphics card, or some other non-AMD display device, I would assume.

**Comment by kentrussell** [comment]
Fixed in 2.2
